### DISCUSSION TITLE
F5 - Onnx - Piper

### DISCUSSION THREAD

**Discussion by prevoste** [SWivid/F5-TTS#625] [Q&A]
Hi all,

I have installed f5 on a Raspberry Pi 5 and it generates very good quality output but to be expected it is very slow. I am trying to understand how f5 works, does it take a standard model and modify it in some way using the ref_text & audio before generating the desired output (gen_text)? Is there an intermediate stage that could be executed separately?

Is there any way to generate an onnx file from the initial model and the provided sample audio so that it could be used with piper to generate an audio output? 

 Thanks

**Comment by SWivid** [comment]
check for readme and issues, there is onnx version
